File: PathoFroh/src/main/java/com/patho/main/repository/jpa/ListItemRepository.java
package com.patho.main.repository.jpa;

import com.patho.main.model.system.ListItem;
import com.patho.main.model.system.ListItemType;
import com.patho.main.repository.jpa.custom.ListItemRepositoryCustom;

import java.util.List;

public interface ListItemRepository extends BaseRepository<ListItem, Long>, ListItemRepositoryCustom {
    public List<ListItem> findByListTypeAndArchivedOrderByIndexInListAsc(ListItemType list, boolean archived);
}

File: PathoFroh/src/main/kotlin/com/patho/main/repository/jpa/ListItemRepository.kt
package com.patho.main.repository.jpa

import com.patho.main.model.preset.ListItem
import com.patho.main.model.preset.ListItemType
import com.patho.main.model.preset.MaterialPreset
import com.patho.main.repository.jpa.custom.ListItemRepositoryCustom

interface ListItemRepository :  BaseRepository<ListItem, Long>, ListItemRepositoryCustom {
    fun findByListTypeAndArchivedOrderByIndexInListAsc(list: ListItemType, archived: Boolean): List<ListItem>
}